drivers/flash/soc_flash_nrf.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SOC_FLASH_NRF_H__
+#define __SOC_FLASH_NRF_H__
+
+#include <kernel.h>
+
+#define FLASH_OP_DONE    (0) /* 0 for compliance with the driver API. */
+#define FLASH_OP_ONGOING (-1)
+
+struct flash_context {
+	uint32_t data_addr;  /* Address of data to write. */
+	uint32_t flash_addr; /* Address of flash to write or erase. */
+	uint32_t len;        /* Size off data to write or erase [B]. */
+#ifndef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE
+	uint8_t  enable_time_limit; /* set execution limited to the execution
+				     * window.
+				     */
+	void     *synch_context;    /* timing context used by given
```

Comment:
I was thinking if such synchronization backend context could be maintained within synchronization backend module and not exposed to `soc_flash_nrf` at all. It would require changes in `nrfsb_get_write_synch_context()` and `nrfsb_get_erase_synch_context()` declaration to return `void` instead of `void*`; and removal of `void *synch_context` from this structure.

I think it would simplify both sync backends and this module.

Reply:
I agree with you.